File: android/src/main/kotlin/com/rokt/rokt_sdk/RoktWidgetFactory.kt
package com.rokt.rokt_sdk

import android.content.Context
import com.rokt.roktsdk.Widget
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class RoktWidgetFactory(private val messenger: BinaryMessenger): PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    val nativeViews = mutableMapOf<Int, Widget>()

    override fun create(context: Context?, viewId: Int, args: Any?): PlatformView {
        val roktWidget = RoktWidget(context!!, messenger, viewId)
        nativeViews[viewId] = roktWidget.widget
        return roktWidget
    }
}

File: ios/Classes/RoktWidgetFactory.swift
//
//  RoktWidgetFactory.swift
//  rokt_sdk
//
//  Copyright 2020 Rokt Pte Ltd
//
//  Licensed under the Rokt Software Development Kit (SDK) Terms of Use
//  Version 2.0 (the "License");
//
//  You may not use this file except in compliance with the License.
//
//  You may obtain a copy of the License at https://rokt.com/sdk-license-2-0/

import Flutter
import UIKit
import Rokt_Widget

class RoktWidgetFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger
    var platformViews: [FLRoktWidgetView] = []
    
    init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
        super.init()
    }
    
    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        let platformView = FLRoktWidgetView(
            frame: frame,
            viewIdentifier: viewId,
            arguments: args,
            binaryMessenger: messenger)
        platformViews.append(platformView)
        return platformView
    }
}
